quakemigrate/plot/event.py

```diff
@@ -173,34 +172,42 @@ def _plot_waveform_gather(ax, lut, phases, event, idx):
     # Handle single-phase plotting
     pick_colours = PICK_COLOURS
     if len(phases) == 1:
-        if phases[0] == "S":
-            pick_colours = [PICK_COLOURS[1]]
+        if phases[0] == "P":
+            pick_colours = [PICK_COLOURS[0]]
     for arrival, c, phase in zip(arrivals, pick_colours, phases):
         ax.scatter(arrival, range_order, s=s, c=c, marker="|", zorder=5,
                    lw=1.5, label=f"Modelled {phase}")
 
     # --- Waveforms ---
-    times_utc = event.data.filtered_waveforms[0].times(type="UTCDateTime")
-    mint, maxt = event.otime - 0.1, event.otime + np.max(traveltimes)*1.5
+    waveforms = event.onset_data.filtered_waveforms
+    # Min and max times to plot
+    mint = event.otime - 0.1
+    maxt = min(event.otime + np.max(traveltimes)*1.5, event.data.endtime)
+    # Convert to indices -- will still be the same for sub-sample shifts
+    times_utc = waveforms[0].times("UTCDateTime")
     mint_i, maxt_i = [np.argmin(abs(times_utc - t)) for t in (mint, maxt)]
-    times_plot = event.data.filtered_waveforms[0].times(type="matplotlib")[mint_i:maxt_i]
+    # Loop through stations
```

Comment:
Redundant, the next line is pretty explicit.

Reply:
```suggestion

```